### PR TITLE
TILA-584 Api to get reservation unit calendar urls.

### PR DESCRIPTION
**Pull request by aprenfors**
Application event ical url and reservation unit ical url.

### CODE REVIEW

File: api/ical_api.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        )
+        buffer.seek(0)
+
+        return FileResponse(buffer, as_attachment=True, filename="file.ics")
+
+
+def reservation_unit_calendar(reservation_unit: ReservationUnit):
+    cal = Calendar()
+
+    cal.add("version", "2.0")
```
Comment (nicobav) [inline api/ical_api.py:130]: should the version come from some variable / settings?

File: api/ical_api.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    reservations = Reservation.objects.filter(
+        recurring_reservation__application_event=application_event
+    )
+    for reservation in reservations:
+        ical_event = Event()
+        ical_event.add("summary", application_event.name)
+        ical_event.add("dtstart", reservation.begin)
+        ical_event.add("dtend", reservation.end)
+        ical_event.add("dtstamp", datetime.datetime.now() + datetime.timedelta(hours=1))
```
Comment (nicobav) [inline api/ical_api.py:182]: why this timestamp needs to be one hour ahead of current time?

File: tilavarauspalvelu/settings.py [hunk 3]
```diff
@@ -169,6 +169,7 @@ def get_git_revision_hash() -> str:
     HAUKI_ORIGIN_ID=(str, "tvp"),
     CSRF_TRUSTED_ORIGINS=(list, []),
     MULTI_PROXY_HEADERS=(bool, False),
+    ICAL_HASH_SECRET=(str, "qhoew923uqqwee"),
```
Comment (nicobav) [inline tilavarauspalvelu/settings.py:172]: perhaps just return empty if not in env.

File: applications/models.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import datetime
 import logging
 import math
+import uuid as uuid
```
Comment (nicobav) [inline applications/models.py:4]: `import uuid`

File: api/ical_api.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+from rest_framework.reverse import reverse
+from rest_framework.viewsets import ViewSet
+
+from applications.models import ApplicationEvent
+from permissions.api_permissions import ReservationUnitCalendarUrlPermission
+from reservation_units.models import ReservationUnit
+from reservations.models import Reservation
+
+
+class ExampleSerializer(serializers.ModelSerializer):
```
Comment (nicobav) [inline api/ical_api.py:22]: :construction_worker_man:

**Review by nicobav — CHANGES_REQUESTED** [review]

File: api/ical_api.py (hunk 5 above)
Comment (aprenfors) [reply api/ical_api.py:22]: 🤦

File: api/ical_api.py (hunk 1 above)
Comment (aprenfors) [reply api/ical_api.py:130]: I guess it could be good.

File: api/ical_api.py (hunk 2 above)
Comment (aprenfors) [reply api/ical_api.py:182]: It shouldn't, good catch. No idea why i put it like this.

**Comment by aprenfors** [comment]
Wonder why that test started failing... Shouldn't really be anything in this that changes it, but oh well.

File: api/ical_api.py (hunk 1 above)
Comment (aprenfors) [reply api/ical_api.py:130]: Setting might be a bit heavy handed, maybe just a constant would be better.

**Review by nicobav — APPROVED** [review]
:ship:

**Pull request merged**